warehouse/app/warehouse/sftp.py

```diff
@@ -157,7 +157,23 @@ def readlink(self, path):
         return sftp.SFTP_OP_UNSUPPORTED
 
     def remove(self, path):
-        return sftp.SFTP_OP_UNSUPPORTED
+
+        path = self.canonicalize(path)
+
+        # check permission
+        if not path.startswith(self.root):
+            return sftp.SFTP_PERMISSION_DENIED
+
+        # return error if file doesn't exists
+        if not os.path.exists(path):
+            return sftp.SFTP_FAILURE
+
+        try:
+            os.unlink(path)
```

Comment:
should we consider the case if path is a dir?

Reply:
right.